capellambse_context_diagrams/collectors/element_relation_view.py

```diff
@@ -0,0 +1,183 @@
+        self.right_box = _elkjs.ELKInputChild(
+            id="__HideElement:right",
+            layoutOptions=_elkjs.get_global_layered_layout_options(),
+            children=[],
+            edges=[],
+        )
+        self.left_boxes_n = 0
+        self.global_boxes: dict[str, _elkjs.ELKInputChild] = {}
+        self.classes: dict[str, information.Class] = {}
+
+    def process(self) -> None:
+        for item in self.diagram.target.allocated_exchange_items:
+            if not (parent_box := self.global_boxes.get(item.parent.uuid)):
+                parent_box = self.global_boxes.setdefault(
+                    item.parent.uuid,
+                    makers.make_box(
+                        item.parent,
+                        layout_options=makers.DEFAULT_LABEL_LAYOUT_OPTIONS,
+                    ),
+                )
+                self.left_box.children.append(parent_box)
+
+            if item.uuid in (i.id for i in parent_box.children):
+                continue
+            box = makers.make_box(item)
+            parent_box.children.append(box)
+            self.left_boxes_n += 1
+
+            for elem in item.elements:
+                if elem.abstract_type:
+                    self.classes.setdefault(
+                        elem.abstract_type.uuid, elem.abstract_type
+                    )
+                    eid = f"__ExchangeItemElement:{item.uuid}_{elem.abstract_type.uuid}"
+                    self.data.edges.append(
+                        _elkjs.ELKInputEdge(
+                            id=eid,
+                            layoutOptions=DEFAULT_EDGE_LAYOUT_OPTIONS,
+                            sources=[item.uuid],
+                            targets=[elem.abstract_type.uuid],
+                            # Add back labels once edge label issue is fixed
+                            # labels=makers.make_label(elem.name),
+                        )
+                    )
+        for cls in self.classes.values():
+            box = self._make_class_box(cls)
+            partition = 0
+            current = cls
+            while current.super and current.super.uuid in self.classes:
+                partition += 1
+                current = current.super
+            box.layoutOptions["partitioning.partition"] = partition
+            self.tree_view_box.children.append(box)
+            if partition > 0:
+                eid = f"__Generalization:{cls.super.uuid}_{cls.uuid}"
+                self.tree_view_box.edges.append(
+                    _elkjs.ELKInputEdge(
+                        id=eid,
+                        sources=[cls.super.uuid],
+                        targets=[cls.uuid],
+                    )
+                )
+
+        right_boxes_n = 0
+        while self.left_boxes_n > right_boxes_n:
+            box = self.left_box.children.pop()
+            self.right_box.children.append(box)
+            n = len(box.children)
+            self.left_boxes_n -= n
+            right_boxes_n += n
```

Comment:
If you just evaluate the length of the left and right boxes when ever you need it, you don't need to manually update the length variables `left_boxes_n` and `right_boxes_n`. I mean you have `self.left_box` and `self.right_box` on which you can easily check length of their children.
Then you just  need to `pop` and `append` here.